src/sagemaker/model.py

```diff
@@ -592,7 +592,12 @@ def prepare_container_def(
         deploy_env = copy.deepcopy(self.env)
         if self.source_dir or self.dependencies or self.entry_point or self.git_config:
             is_repack = (
-                self.source_dir and self.entry_point and not (self.key_prefix or self.git_config)
+                self.source_dir
+                and self.entry_point
+                and self.model_data
```

Comment:
sanity-check: are you sure `self.model_data` is always required?

For example, could there be cases where the script gets passed as `file://` (from local file system) with no model S3 URI ?

Reply:
actually, good catch. i didn't intend to commit that. let me re-run the unit tests to confirm...